### ISSUE TITLE
Hovering on the video doesn't play the video. Clicking on the video opens a new tab

### ISSUE THREAD

**Issue by sinthujan24** [cypress-io/cypress#14757]
### Current behavior

1) Hovering on a video tile doesn't play the video as expected
2) Clicking on the video opens a new tab (Expected to open the video player on the same screen) 

### Desired behavior

1) Hovering on a video should play the video 
2) Clicking on the video should open up the video player and video should play 

### Test code to reproduce

1) **Hovering on a video should play the video** 
```js
const urls = [`https://cat-press.com/cat-movies/firework`]
  urls.forEach((url) => {
    it(`Should load the videos and autoplay one: ${url}`, () => {
      cy.visit(url)
      // Scroll to feed
      cy.get('.cy-grid').scrollIntoView()
      // Should hover over to the video at index 1
      cy.get(`.cy-thumbnail-link`).eq(1).trigger('mouseover')
     
    })
  })
```

2) Clicking on the video opens a new tab (Expected to open the video player on the same screen) 

```js
const urls = [`https://cat-press.com/cat-movies/firework`]
  urls.forEach((url) => {
    it(`Should load the videos and autoplay one: ${url}`, () => {
      cy.visit(url)
      // Scroll to feed
      cy.get('.cy-grid').scrollIntoView()
      // Should hover over to the video at index 1
      cy.get(`.cy-thumbnail-link`).eq(1).trigger('click')
     
    })
  })
```

![image](https://user-images.githubusercontent.com/76002992/105928410-e48ca600-6013-11eb-8478-fd6865dbed3b.png)

### Versions

Mac: Catalina 10.15.7
Chrome: Version 87.0.4280.141 (Official Build) (x86_64)
Cypress: Version 6.1.0 (6.1.0.531243)

**Comment by bahmutov** [comment]
1. Cypress does not have a native hover yet https://github.com/cypress-io/cypress/issues/311, try using https://github.com/dmtrKovalenko/cypress-real-events
2. figure out why and how the app determines how to open the window. Some points in https://glebbahmutov.com/blog/cypress-tips-and-tricks/#deal-with-target_blank and https://glebbahmutov.com/blog/cypress-tips-and-tricks/#deal-with-windowopen but it could also be due to the iframe app, try using `experimentalSourceRewriting: true` https://on.cypress.io/experimental

**Comment by sinthujan24** [comment]
@bahmutov  2. Manually it doesn't open a second tab. It remains on the same page.

**Comment by bahmutov** [comment]
Try the experimental source rewriting. It probably has to do with the app being inside the iframe

**Comment by sinthujan24** [comment]
> Try the experimental source rewriting. It probably has to do with the app being inside the iframe
> […](#)
> Sent from my iPhone
> On Jan 27, 2021, at 13:43, sinthujan24 ***@***.***> wrote: ﻿ @bahmutov 2. Manually it doesn't open a second tab. It remains on the same page. — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub, or unsubscribe.

@bahmutov I tried it. it still opens a new tab.

**Comment by sinthujan24** [comment]
I kindly ask you to reopen this issue:

1) it's not a request to support multi tabs. Rather, cypress behaves differently than manual user interactions. Here, the tab is opened unexpectedly

**Comment by bahmutov** [comment]
So I created reproduction in https://github.com/cypress-io/cypress-test-tiny/tree/opens-new-tab

In the normal browser I see a click event handler attached to those video cards

<img width="1440" alt="Screen Shot 2021-01-27 at 2 16 04 PM" src="https://user-images.githubusercontent.com/2212006/106041989-7e069700-60aa-11eb-859d-d10273dc8b6c.png">

In Cypress the same element does not have an event handler.

<img width="1276" alt="Screen Shot 2021-01-27 at 2 16 48 PM" src="https://user-images.githubusercontent.com/2212006/106042022-8ced4980-60aa-11eb-90db-d8e02c774aff.png">

The site has bunch of errors while loading. If you can eliminate everything from the page to find out how the event listener is attached in the browser and why it is not attached inside Cypress, I would be happy to reopen the issue. But without this investigation by someone who understands the implementation of those video cards, I don't think it is productive to investigate.

**Comment by sinthujan24** [comment]
@bahmutov I'm seeing element has an event handler and it's click. Image attached

![image](https://user-images.githubusercontent.com/76002992/106050546-4fda8480-60b5-11eb-9c13-725e9310b2a6.png)

**Comment by bahmutov** [comment]
Very weird

**Comment by sinthujan24** [comment]
> Very weird
> […](#)
> Sent from my iPhone
> On Jan 27, 2021, at 15:37, sinthujan24 ***@***.***> wrote: ﻿ @bahmutov I'm seeing element has an event handler and it's click. Image attached — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub, or unsubscribe.

Yes. And Unfortunately, hover and clicking on the video tile to open the player are the most critical part of our testing. That's why I've been struggling for a solution for last couple of days. Please reopen this issue if the above evidences earns an investigation on cypress side at the least

**Comment by bahmutov** [comment]
Can you extract the minimal reproducible example from this site?

**Comment by sinthujan24** [comment]
@bahmutov The code snippets mentioned in https://github.com/cypress-io/cypress/issues/14757#issue-794685850 are the minimal reproducible steps. 

Did you want me to commit this to https://github.com/cypress-io/cypress-test-tiny/tree/opens-new-tab

**Comment by bahmutov** [comment]
The problem with the example is that it is a complicated site with console errors that might or might not matter. If you had a small page with those video cards and they did not work correctly I would appreciate it. On the current media url it is hard to determine the problem

**Comment by sinthujan24** [comment]
@bahmutov Please use the below example
**Click Issue**

```
const urls = [`https://fireworktv.com/embed/demo`]
  urls.forEach((url) => {
    it(`Should load the videos and autoplay one: ${url}`, () => {
      cy.visit(url)
      cy.get(`.cy-thumbnail-link`).eq(1).trigger('click')
    })
  })
```

**Hover issue:** 
```
const urls = [`https://fireworktv.com/embed/demo`]
  urls.forEach((url) => {
    it(`Should load the videos and autoplay one: ${url}`, () => {
      cy.visit(url)
      cy.get(`.cy-thumbnail-link`).eq(1).trigger('mouseover')
    })
  })
```

**Comment by bahmutov** [comment]
I have updated the reproducible example in https://github.com/cypress-io/cypress-test-tiny/tree/opens-new-tab

There is a click on the video player that is handled by Preact which I cannot figure out where it is going, since this is minified production code. Everyone is welcome to trace it and figure out why the behavior during testing is different.

**Comment by sinthujan24** [comment]
> ### Current behavior
> 1. Hovering on a video tile doesn't play the video as expected
> 2. Clicking on the video opens a new tab (Expected to open the video player on the same screen)
> 
> ### Desired behavior
> 1. Hovering on a video should play the video
> 2. Clicking on the video should open up the video player and video should play
> 
> ### Test code to reproduce
> 1. **Hovering on a video should play the video**
> 
> ```js
> const urls = [`https://cat-press.com/cat-movies/firework`]
>   urls.forEach((url) => {
>     it(`Should load the videos and autoplay one: ${url}`, () => {
>       cy.visit(url)
>       // Scroll to feed
>       cy.get('.cy-grid').scrollIntoView()
>       // Should hover over to the video at index 1
>       cy.get(`.cy-thumbnail-link`).eq(1).trigger('mouseover')
>      
>     })
>   })
> ```
> 
> 1. Clicking on the video opens a new tab (Expected to open the video player on the same screen)
> 
> ```js
> const urls = [`https://cat-press.com/cat-movies/firework`]
>   urls.forEach((url) => {
>     it(`Should load the videos and autoplay one: ${url}`, () => {
>       cy.visit(url)
>       // Scroll to feed
>       cy.get('.cy-grid').scrollIntoView()
>       // Should hover over to the video at index 1
>       cy.get(`.cy-thumbnail-link`).eq(1).trigger('click')
>      
>     })
>   })
> ```
> 
> ![image](https://user-images.githubusercontent.com/76002992/105928410-e48ca600-6013-11eb-8478-fd6865dbed3b.png)
> 
> ### Versions
> Mac: Catalina 10.15.7
> Chrome: Version 87.0.4280.141 (Official Build) (x86_64)
> Cypress: Version 6.1.0 (6.1.0.531243)

1. The first issue is fixed by specifiying an explicit timeout after hover operation.

```
// Click on first thumbnail
      cy.get(`.cy-thumbnail-link`)
        .eq(1)
        .trigger("mouseover")
        //.trigger("mouseenter")
      cy.wait(3000)
```
